File: 2402/dayeong/ponkemon.java
import java.util.HashSet;
import java.util.Set;
import java.util.Arrays;
class Solution {
    public int solution(int[] nums) {
       int max =  Arrays.stream(nums).boxed()//스트림 생성
            .distinct().count();
        return answer;
    }
}

import java.util.Arrays;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
       String[] a =phone_book;
        int cnt =phone_book.length;
        for(int i =0 ; i<cnt;i++)
        {
            if((int)Arrays.stream(phone_book).filter(x->(x==a[i])).count() <= 2)
            {
                answer=false;
                break;
            }
           
        }
        return answer;
    }
}

File: 2403-1:kyoungho/Ponkemon.py
# Ponkemon.py
# 2점 정답
def solution(nums):
    answer = 0
    
    unique_list = list(set(nums))   ## 중복값 제거
    
    ## 최대 종류수는 배열의 길이 or 숫자의 종류 갯수 중 작은거
    return min(len(unique_list), len(nums)//2)
